Libraries/Opc.Ua.Security.Certificates/Extensions/X509AuthorityKeyIdentifierExtension.cs

```diff
@@ -0,0 +1,340 @@
+
+        /// <summary>
+        /// The alternate OID for a Authority Key Identifier extension.
+        /// </summary>
+        public const string AuthorityKeyIdentifier2Oid = "2.5.29.35";
+
+        /// <summary>
+        /// The identifier for the key as a little endian hexadecimal string.
+        /// </summary>
+        public string KeyIdentifier => m_keyIdentifier.ToHexString();
+
+        /// <summary>
+        /// The identifier for the key as a byte array.
+        /// </summary>
+        public byte[] GetKeyIdentifier() => m_keyIdentifier;
+
+        /// <summary>
+        /// A list of distinguished names for the issuer.
+        /// </summary>
+        public X500DistinguishedName Issuer => m_issuer;
+
+        /// <summary>
+        /// The serial number of the authority key as a big endian hexadecimal string.
+        /// </summary>
+        public string SerialNumber => m_serialNumber.ToHexString(true);
+
+        /// <summary>
+        /// The serial number of the authority key as a byte array in little endian order.
+        /// </summary>
+        public byte[] GetSerialNumber() => m_serialNumber;
+        #endregion
+
+        #region Private Methods
+        private byte[] Encode()
+        {
+            AsnWriter writer = new AsnWriter(AsnEncodingRules.DER);
+            writer.PushSequence();
+
+            if (m_keyIdentifier != null)
+            {
+                Asn1Tag keyIdTag = new Asn1Tag(TagClass.ContextSpecific, 0);
+                writer.WriteOctetString(m_keyIdentifier, keyIdTag);
+            }
+
+            if (m_issuer != null)
+            {
+                Asn1Tag issuerNameTag = new Asn1Tag(TagClass.ContextSpecific, 1);
+                writer.PushSequence(issuerNameTag);
+
+                // Add the issuer to constructed context-specific 4 (GeneralName.directoryName)
+                Asn1Tag directoryNameTag = new Asn1Tag(TagClass.ContextSpecific, 4, true);
+                writer.PushSetOf(directoryNameTag);
+                writer.WriteEncodedValue(m_issuer.RawData);
+                writer.PopSetOf(directoryNameTag);
+                writer.PopSequence(issuerNameTag);
+            }
+
+            if (m_serialNumber != null)
+            {
+                Asn1Tag issuerSerialTag = new Asn1Tag(TagClass.ContextSpecific, 2);
+                BigInteger issuerSerial = new BigInteger(m_serialNumber);
+                writer.WriteInteger(issuerSerial, issuerSerialTag);
+            }
+
+            writer.PopSequence();
+            return writer.Encode();
+        }
+
+
+        private void Decode(byte[] data)
+        {
+            if (base.Oid.Value == AuthorityKeyIdentifierOid ||
+                base.Oid.Value == AuthorityKeyIdentifier2Oid)
+            {
+                try
+                {
+                    AsnReader dataReader = new AsnReader(data, AsnEncodingRules.DER);
+                    var akiReader = dataReader?.ReadSequence();
```

Comment:
`new` can't return null, so the `?` in `?.` is redundant.

```suggestion
                    var akiReader = dataReader.ReadSequence();
```

Also, since you don't use `var` anywhere else, but on this line the return type isn't obvious, it's odd that you chose to use it here.

Reply:
done in all places